Library/Homebrew/test/cmd/install_spec.rb

```diff
@@ -72,4 +72,14 @@ def install
       .and be_a_success
     expect(HOMEBREW_CELLAR/"testball1/HEAD-d5eb689/foo/test").not_to be_a_file
   end
+
+  it "installs formulae with debug symbols", :integration_test do
+    setup_test_formula "testball1"
+
+    expect { brew "install", "testball1", "--debug-symbols", "-s" }
+      .to output(%r{#{HOMEBREW_CELLAR}/testball1/0\.1}o).to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+    expect(HOMEBREW_CELLAR/"testball1/0.1/foo/test").not_to be_a_file
```

Comment:
Maybe worth checking there are some debug files written somewhere?

Reply:
I'm unable to figure out how to do this. `byebug` does not work inside `brew tests` despite the `--byebug` switch, I think because it's running inside its own process. So I couldn't stop the test and examine the file system. The only way I could get output out of the spec was to `raise` with the text I wanted to find. So I did that, but I couldn't actually find any files inside the spec context so I don't know where they are going or how to examine/test their presence. There are also no positive test examples I could find where this was happening.

Reply:
After much trial and error, done in fdf17f06b1c9a35e544653ed4c46ea0b9866e610